### PR TITLE
fix bash rprompt

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

- refactor: remove shell name from plain init
- fix(bash): escape rprompt entirely

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by JanDeDobbeleer** [comment]
If this works, it needs a serious cleanup.

File: src/engine/engine.go [hunk 1]
```diff
@@ -133,22 +133,26 @@ func (e *Engine) renderBlock(block *Block) {
...
+				ansi = &color.Ansi{}
+				ansi.InitPlain()
+				e.writeANSI(`\[`)
+			}
+			e.writeANSI(ansi.CarriageForward())
+			e.writeANSI(ansi.GetCursorForRightWrite(length, block.HorizontalOffset))
 			e.currentLineLength = 0
 			e.write(text)
+			if e.Env.Shell() == shell.BASH {
+				e.writeANSI(`\[`)
```
Comment (akinomyoga) [inline src/engine/engine.go]: Now the right prompt is enclosed within `\[...\[` in `PS1`.

```suggestion
				e.writeANSI(`\]`)
```

I think this is a typo.

**Comment by akinomyoga** [comment]
Thank you! This also works!

**Comment by JanDeDobbeleer** [comment]
@akinomyoga I need to double check one thing and this should be good to go. Sorry for the delay, started a new job.

**Comment by akinomyoga** [comment]
Thanks! I've confirmed that the new version also works!

**Pull request merged**